js/ilMumieTaskForm.js

```
@@ -33,75 +28,52 @@
-             * @param {Object} course
+             * Update the hidden input field with the selected course's course file path
              */
-            function addOptionForCourse(course) {
-                const optionCourse = document.createElement("option");
-                const selectedLanguage = langController.getSelectedLanguage();
-                let name;
-                // If the currently selected server is not available on the server, we need to select another one.
-                if (!course.languages.includes(selectedLanguage)) {
-                    name = course.name[0];
-                } else {
-                    name = course.name.find(n => n.language === selectedLanguage);
-                }
-                optionCourse.setAttribute("value", name.value);
-                optionCourse.text = name.value;
-                courseDropDown.append(optionCourse);
+            function updateCoursefilePath(courseFile) {
+                coursefileElem.value = courseFile;
+                updateCourseName();
             }
 
             /**
-             * Update the hidden input field with the selected course's course file path
+             * Update displayed course name.
              */
-            function updateCoursefilePath() {
-                coursefileElem.value = courseController.getSelectedCourse().path_to_course_file;
+            function updateCourseName() {
+                const selectedCourse = courseController.getSelectedCourse();
+                const selectedLanguage = langController.getSelectedLanguage();
+                if (!selectedCourse || !selectedLanguage) {
+                    return;
+                }
+                const name = selectedCourse.name
+                .find(translation => translation.language === selectedLanguage)?.value;
+                courseNameElement.value = name;
+                courseNameDisplayElement.value = name;
```

Negate the boolean expression used for the "if" statement and execute the code below the statement accordingly, then you have also fail fast and don't need  the "return" statement

---

done